ragna/core/_rag.py

```diff
@@ -148,8 +153,10 @@ def __init__(
         self._rag = rag
 
         self.documents = self._parse_documents(documents)
-        self.source_storage = self._rag._load_component(source_storage)
-        self.assistant = self._rag._load_component(assistant)
+        self.source_storage = cast(
+            SourceStorage, self._rag._load_component(source_storage)
+        )
+        self.assistant = cast(Assistant, self._rag._load_component(assistant))
```

Comment:
This is one of the situations where Python is too dynamic to statically type. The return type is only `typing.Optional[SourceStorage]` when we call `_load_component(..., ignore_unavailable=True)`. Without that, i.e. what we do here, the return type is `SourceStorage`.

I we remove the `cast`, i.e. annotating as `typing.Optional[SourceStorage]` we get all sorts of problems down the line, because `mypy` will (correctly) complain that the object might be `None` and you can't call methods on it etc.

Reply:
I see, make sense. Thanks for explanation.